### PR TITLE
Add pre-commit support

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
This PR adds a file so that tslint can be used natively with [pre-commit](http://pre-commit.com/).

If this PR is accepted, I'll submit a PR to update the documentation on pre-commit's side too!

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#     sha: e3200b7873d03e7b13b7f8d0bd5cb760ed400585
+#     hooks:
+#     -   id: tsfmt
+
+-   id: tsfmt
+    name: TypeScript Formatter (tsfmt)
+    description: TypeScript code auto-formatter.
+    entry: tsfmt
+		args: ['--verify']
+    language: system
```
Comment (vvakame) [inline .pre-commit-hooks.yaml]: Why do you use `system`?
Is `node` more appropriate? http://pre-commit.com/#node

My suggestions, We should use project local tsfmt, not system wide tsfmt.
I think you have a two or more project about tsfmt using. There project should be able to update dependencies independently.

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    files: \.ts$
```
Comment (vvakame) [inline .pre-commit-hooks.yaml]: Is this the regexp?
I want to use `\.tsx?$`.

**Comment by vvakame** [comment]
👀  https://github.com/palantir/tslint/pull/2562

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (alexjurkiewicz) [inline .pre-commit-hooks.yaml]: good point. I haven't used `node` language before, but it seems like tsfmt should be compatible as-is.

File: .pre-commit-hooks.yaml (hunk 2 above)
Comment (alexjurkiewicz) [inline .pre-commit-hooks.yaml]: done

**Comment by vvakame** [comment]
I tried to use pre-commit, but I can't do it..
https://github.com/vvakame/typescript-formatter/tree/alexjurkiewicz-pre-commit
https://gist.github.com/vvakame/3ae719362d4de86b966ae6f79deb0d58

I have a question.
tsfmt & tslint projects are must be installed via npm.
Is pre-commit project can do that? I think that that project is not support it well...

**Comment by alexjurkiewicz** [comment]
Sorry for the error. I thought I understood pre-commit + node but I didn't 🙇 .

The solution is to add an `additional_dependencies` key to `.pre-commit-hooks.yaml`:
```yaml
-   id:          tsfmt
    name:        TypeScript Formatter (tsfmt)
    description: TypeScript code auto-formatter.
    entry:       tsfmt --verify
    language:    node
    files:       \.tsx?$
    additional_dependencies: ['typescript-formatter', 'typescript']
```

You can also specify a certain version of dependencies like using npm/yarn: 
```yaml
additional_dependencies: ['typescript-formatter', 'typescript@2.2.2']
```

However, this will mean that there would need to be a new commit for each typescript version that changes typescript-formatter behaviour. Each update to the pre commit file could be recorded for users to use a certain version of typescript with. Example `.pre-commit-hooks.yaml` content:

```yaml
# Use the following SHAs:
# * typescript 1.9: aaaaaaaa
# * typescript 2.0: bbbbbbb
# * typescript 2.2: cccccccc
-   repo: git@github.com:vvakame/typescript-formatter.git
    sha: <sha>
[...]
```

This is getting a bit complex, sorry. I think you wouldn't like this in your main repo, is that right?

What I can do instead is maintain a dedicated pre-commit-tsfmt repository, like [jshint](https://github.com/pre-commit/mirrors-jshint).

**Comment by alexjurkiewicz** [comment]
One possible additional complication. The `additional_dependencies` line should probably specify the typescript-formatter version.

**Comment by vvakame** [comment]
sorry for late reply.

> I think you wouldn't like this in your main repo, is that right?
> What I can do instead is maintain a dedicated pre-commit-tsfmt repository, like jshint.

sounds good! I recommended it.
I am not a user, so it would be better to maintain it by the people actually using it.

**Pull request closed without merging**